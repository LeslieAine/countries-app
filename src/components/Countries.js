import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchCountries } from '../redux/countries/countriesSlice';
import Country from './Country';
import styles from '../styles/Countries.module.css';

const Countries = () => {
  const dispatch = useDispatch();
  const countries = useSelector((state) => state.countries);

  const [continent, setContinent] = useState('all');

  useEffect(() => {
    if (countries.length === 0) {
      dispatch(fetchCountries());
    }
  }, [countries, dispatch]);

  return (
    <div>
      <h1 className={styles.title}>Countries</h1>
      <div className={styles.dropdown}>
        <select
          name="continents"
          id="continents"
          className={styles.select}
          onChange={(e) => setContinent(e.target.value)}
        >
          <option value="all">All</option>
          <option value="Africa">Africa</option>
          <option value="Asia">Asia</option>
          <option value="Europe">Europe</option>
          <option value="North America">North America</option>
          <option value="Oceania">Oceania</option>
          <option value="South America">South America</option>
          <option value="Antarctica">Antarctica</option>
        </select>
      </div>
      <ul className={styles.countries}>
        {countries.filter((country) => {
          if (continent === 'all') {
            return country;
          }
          return country.continent === continent;
        }).map((country) => (
          <Country
            key={country.name}
            name={country.name}
            continent={country.continent}
            flagEmoji={country.flagEmoji}
          />
        ))}
      </ul>
    </div>
  );
};

export default Countries;
