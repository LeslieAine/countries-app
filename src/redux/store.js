import countriesReducer from './countries/countriesSlice';

const { configureStore } = require('@reduxjs/toolkit');

const store = configureStore({
  reducer: {
    countries: countriesReducer,
  },
});

export default store;
