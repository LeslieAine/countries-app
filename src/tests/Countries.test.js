import React from 'react';
import { Provider } from 'react-redux';
import { screen, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import store from '../redux/store';
import Countries from '../components/Countries';

describe('Countries', () => {
  it('should render the component', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Countries />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have a select element', () => {
    render(
      <Provider store={store}>
        <Countries />
      </Provider>,
    );
    expect(screen.getByRole('combobox')).toBeTruthy();
  });

  it('should have a select element with 8 options', () => {
    render(
      <Provider store={store}>
        <Countries />
      </Provider>,
    );
    expect(screen.getAllByRole('option').length).toBe(8);
  });
});
