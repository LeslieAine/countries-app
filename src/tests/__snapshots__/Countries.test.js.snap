// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countries should render the component 1`] = `
<div>
  <h1
    className="title"
  >
    Countries
  </h1>
  <div
    className="dropdown"
  >
    <select
      className="select"
      id="continents"
      name="continents"
      onChange={[Function]}
    >
      <option
        value="all"
      >
        All
      </option>
      <option
        value="Africa"
      >
        Africa
      </option>
      <option
        value="Asia"
      >
        Asia
      </option>
      <option
        value="Europe"
      >
        Europe
      </option>
      <option
        value="North America"
      >
        North America
      </option>
      <option
        value="Oceania"
      >
        Oceania
      </option>
      <option
        value="South America"
      >
        South America
      </option>
      <option
        value="Antarctica"
      >
        Antarctica
      </option>
    </select>
  </div>
  <ul
    className="countries"
  />
</div>
`;
